bot/src/main/java/com/community/tools/service/openai/OpenAiService.java

```diff
@@ -0,0 +1,121 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OpenAiService {
+  @Value("${openai.url}")
+  private String url;
+  @Value("${openai.token}")
+  private String apiKey;
+  @Value("${openai.model}")
+  private String model;
+  private final ObjectMapper objectMapper;
+
+  public OpenAiService(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * This is the main method in service which processes prompt.
+   * @param prompt users request to the openai api
+   * @return ready-to-use string response from AI
+   */
+  public String processPrompt(String prompt) {
+    try {
+      HttpURLConnection connection = createConnection();
+      sendRequest(connection, prompt);
+      String jsonResponse = readResponse(connection);
+      String raw =  extractMessageFromJsonResponse(jsonResponse);
+      return getStringWithLineSeparators(raw);
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to communicate with OpenAI service", e);
+    }
+  }
+
+  /**
+   * This private method creates connection to api.
+   * @return HttpURLConnection object
+   * @throws IOException IOException
+   */
+  private HttpURLConnection createConnection() throws IOException {
+    URL url = new URL(this.url);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("POST");
+    connection.setRequestProperty("Authorization", "Bearer " + apiKey);
+    connection.setRequestProperty("Content-Type", "application/json");
+    connection.setDoOutput(true);
+    return connection;
+  }
+
+  /**
+   * This private method sends request to openai.
+   * @param connection HttpURLConnection object from createConnection()
+   * @param prompt users request to openai
+   * @throws IOException IOException
+   */
+  private void sendRequest(HttpURLConnection connection, String prompt) throws IOException {
+    OpenAiRequestDto requestDto = new OpenAiRequestDto(model,
+            Collections.singletonList(new OpenAiRequestDto.Message("user", prompt)));
+
+    try (OutputStreamWriter writer = new OutputStreamWriter(connection.getOutputStream())) {
+      String requestBody = objectMapper.writeValueAsString(requestDto);
+      writer.write(requestBody);
+      writer.flush();
+    }
+  }
+
+  /**
+   * This private method gets json response from api.
+   * @param connection HttpURLConnection object from createConnection()
+   * @return json-looking string
+   * @throws IOException IOException
+   */
+  private String readResponse(HttpURLConnection connection) throws IOException {
+    try (BufferedReader bufferedReader = new BufferedReader(
```

Comment:
RestTemplate will also let us avoid this complicated logic of writing/reading request/response data